server/middleware/fetchProfile.js

```diff
@@ -0,0 +1,18 @@
+const Profile = require("../models/Profile");
+
+const fetchProfile = async (req, res, next) => {
+  const userId = req.user.id;
+  if (!userId) {
+    return res.status(401).send("Not authorized");
+  }
+
+  const profile = await Profile.findOne({ userId: req.user.id });
```

Comment:
Alternatively, adding a ref to the Profile within the User model will things a lot easier for us 😄

Reply:
Why didn't I thought about it 😄